mango/processing/processing_time_series.py

```diff
@@ -210,3 +212,218 @@ def create_recurrent_dataset(
+
+    :param df: pd.DataFrame: Pass the dataframe to the function
+    :param date_col: str: Specify the name of the column that contains the date
+    :param as_index: bool: Check if the dataframe is pivoted or not
+    :return: A valueerror if the dataframe is not in the correct format
+    :doc-author: baobab soluciones
+    """
+    if date_col is None and as_index is False:
+        raise ValueError("Dataframe must contain one datetime column")
+    elif date_col is None and as_index:
+        dupli = df.index.duplicated().sum()
+        if dupli > 0:
+            columns_num = sum(
+                pd.api.types.is_numeric_dtype(df[col]) for col in df.columns
+            )
+            if columns_num == len(df.columns):
+                raise ValueError("There are duplicates in the index")
+            else:
+                data = {
+                    "fecha": pd.date_range("2023-01-01", "2023-01-06"),
+                    "ventas_loc1": [30, 50, 10, 25, 32, 45],
+                    "ventas_loc2": [60, 31, 46, 43, 60, 20],
+                }
+                example = pd.DataFrame(data).set_index("fecha")
+                raise ValueError(
+                    f"Dataframe must be pivot:{print(example.to_markdown())}"
+                )
+        else:
+            columns_num = sum(
+                pd.api.types.is_numeric_dtype(df[col]) for col in df.columns
+            )
+
+            if columns_num != len(df.columns):
+                raise ValueError("Not all columns in Dataframe are numerics")
+    elif type(date_col) == list and len(date_col) > 1:
+        raise ValueError("Dataframe must contain one datetime column")
+    elif type(date_col) == list or type(date_col) == str:
+        if type(date_col) == list:
+            date_col = date_col[0]
+        dupli = df[date_col].duplicated().sum()
+        if dupli > 0:
+            columns_num = sum(
+                pd.api.types.is_numeric_dtype(df[col]) for col in df.columns
+            )
+            if columns_num == len(df.columns) - 1:
+                raise ValueError("There are duplicates in the index")
+            else:
+                data = {
+                    "fecha": pd.date_range("2023-01-01", "2023-01-06"),
+                    "ventas_loc1": [30, 50, 10, 25, 32, 45],
+                    "ventas_loc2": [60, 31, 46, 43, 60, 20],
+                }
+                example = pd.DataFrame(data).set_index("fecha")
+                raise ValueError(
+                    f"Dataframe must be pivot:{print(example.to_markdown())}"
+                )
+        else:
+            columns_num = sum(
+                pd.api.types.is_numeric_dtype(df[col]) for col in df.columns
+            )
+
+            if columns_num != len(df.columns) - 1:
+                raise ValueError("Not all columns in Dataframe are numerics")
+
+
+def get_corr_matrix_aux(
+    df: pd.DataFrame,
+    years_corr,
+    n_top,
+    threshold,
+    subset: List = None,
+) -> Dict[str]:
```

Comment:
Dict needs two argument. First is the type of the key and second the type of the value. For this case should be as follows:

```python
Dict[str,Dict[str,float]]
```